packages/flutter_tools/test/general.shard/macos/native_assets_test.dart

```diff
@@ -0,0 +1,318 @@
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/file.dart';
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/artifacts.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/build_info.dart';
+import 'package:flutter_tools/src/build_system/build_system.dart';
+import 'package:flutter_tools/src/features.dart';
+import 'package:flutter_tools/src/globals.dart' as globals;
+import 'package:flutter_tools/src/macos/native_assets.dart';
+import 'package:flutter_tools/src/native_assets.dart';
+import 'package:native_assets_cli/native_assets_cli.dart'
+    hide BuildMode, Target;
+import 'package:native_assets_cli/native_assets_cli.dart' as native_assets_cli;
+import 'package:package_config/package_config_types.dart';
+
+import '../../src/common.dart';
+import '../../src/context.dart';
+import '../../src/fakes.dart';
+
+void main() {
+  late FakeProcessManager processManager;
+  late Environment environment;
+  late Artifacts artifacts;
+  late FileSystem fileSystem;
+  late BufferLogger logger;
+  late Uri projectUri;
+
+  setUp(() {
+    processManager = FakeProcessManager.empty();
+    logger = BufferLogger.test();
+    artifacts = Artifacts.test();
+    fileSystem = MemoryFileSystem.test();
+    environment = Environment.test(
+      fileSystem.currentDirectory,
+      inputs: <String, String>{},
+      artifacts: artifacts,
+      processManager: processManager,
+      fileSystem: fileSystem,
+      logger: logger,
+    );
+    environment.buildDir.createSync(recursive: true);
+    projectUri = environment.projectDir.uri;
+  });
+
+  testUsingContext('dry run with no package config',
+      overrides: <Type, Generator>{
+        ProcessManager: () => FakeProcessManager.any(),
+      }, () async {
+    expect(
+      await dryRunNativeAssetsMacOS(
+        projectUri: projectUri,
+        fileSystem: fileSystem,
+        buildRunner: FakeNativeAssetsBuildRunner(
+          hasPackageConfigResult: false,
+        ),
+      ),
+      null,
+    );
+    expect(
+      (globals.logger as BufferLogger).traceText,
+      contains('No package config found. Skipping native assets compilation.'),
+    );
+  });
+
+  testUsingContext('build with no package config', overrides: <Type, Generator>{
+    ProcessManager: () => FakeProcessManager.any(),
+  }, () async {
+    await buildNativeAssetsMacOS(
+      darwinArchs: <DarwinArch>[DarwinArch.arm64],
+      projectUri: projectUri,
+      buildMode: BuildMode.debug,
+      fileSystem: fileSystem,
+      buildRunner: FakeNativeAssetsBuildRunner(
+        hasPackageConfigResult: false,
+      ),
+    );
+    expect(
+      (globals.logger as BufferLogger).traceText,
+      contains('No package config found. Skipping native assets compilation.'),
+    );
+  });
+
+  testUsingContext('dry run with no package config',
```

Comment:
This test has the same name as the first one in this file, perhaps name it something like "dry run for multiple OSes with no package config"